pkgs/development/haskell-modules/make-package-set.nix

```diff
@@ -221,26 +221,34 @@ in package-set { inherit pkgs stdenv callPackage; } self // {
     #   , overrides : Defaulted (HaskellPackageOverrideSet)
     #   , modifier : Defaulted
     #   , returnShellEnv : Defaulted
+    #   , hoogle : Defaulted
     #   } -> NixShellAwareDerivation
     # Given a path to a haskell package directory, an optional package name
     # which defaults to the base name of the path, an optional set of source
     # overrides as appropriate for the 'packageSourceOverrides' function, an
     # optional set of arbitrary overrides, and an optional haskell package
     # modifier, return a derivation appropriate for nix-build or nix-shell to
-    # build that package.
+    # build that package. By default a hoogle database is generated if a shell
+    # environment is requested.
```

Comment:
There doesn't actually appear to be an explanation here of what `returnShellEnv` does.  Could you add this as well?